pkg/parcacol/querier.go

```diff
@@ -1434,3 +1434,57 @@ func (q *Querier) selectMerge(
 func isSamplesCount(st profile.ValueType) bool {
 	return st.Type == "samples" && st.Unit == "count"
 }
+
+func (q *Querier) GetProfileMetadataMappings(
+	ctx context.Context,
+	query string, startTime, endTime time.Time,
+) ([]string, error) {
+	ctx, span := q.tracer.Start(ctx, "Querier/MappingFiles")
+	defer span.End()
+
+	_, selectorExprs, err := QueryToFilterExprs(query)
+	if err != nil {
+		return nil, err
+	}
+
+	start := timestamp.FromTime(startTime)
+	end := timestamp.FromTime(endTime)
+	filterExpr := logicalplan.And(
+		append(
+			selectorExprs,
+			logicalplan.Col(profile.ColumnTimestamp).GtEq(logicalplan.Literal(start)),
+			logicalplan.Col(profile.ColumnTimestamp).LtEq(logicalplan.Literal(end)),
+		)...,
+	)
+
+	records := []string{}
+	err = q.engine.ScanTable(q.tableName).
+		Filter(filterExpr).
+		Project(logicalplan.Col("stacktrace")).
+		Execute(ctx, func(ctx context.Context, r arrow.Record) error {
+			r.Retain()
+
+			locations := r.Column(0).(*array.List)
+
+			values := locations.ListValues().(*array.Dictionary)
+			valueDict := values.Dictionary().(*array.Binary)
+			for i := 0; i < locations.Len(); i++ {
```

Comment:
thanks to this being dictionary encoded, we can just iterate over the dictionary and already get all unique values